From a2e3926cf70407d05faf44a5b82847dd9c02cca3 Mon Sep 17 00:00:00 2001
From: Alexander Vieth <vieth.alexander@gmx.net>
Date: Thu, 28 Nov 2024 12:45:45 +0100
Subject: [PATCH] dynamic uuid

---
 CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1bc908a..48d4f81 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,6 +51,7 @@ message(STATUS "xeus binary version: v${XEUS_BINARY_VERSION}")
 option(XEUS_BUILD_SHARED_LIBS "Build xeus shared library." ON)
 option(XEUS_BUILD_STATIC_LIBS "Build xeus static library (default if BUILD_SHARED_LIBS is OFF)." ON)
 option(XEUS_STATIC_DEPENDENCIES "link statically with xeus dependencies" OFF)
+option(XEUS_USE_DYNAMIC_UUID  "use dynamic linking for libuuid" OFF)
 
 # Test options
 option(XEUS_BUILD_TESTS "xeus test suite" OFF)
@@ -71,6 +72,7 @@ message(STATUS "XEUS_BUILD_STATIC_LIBS:          ${XEUS_BUILD_STATIC_LIBS}")
 message(STATUS "XEUS_STATIC_DEPENDENCIES:        ${XEUS_STATIC_DEPENDENCIES}")  
 message(STATUS "XEUS_EMSCRIPTEN_WASM_BUILD:      ${EMSCRIPTEN}")
 message(STATUS "XEUS_BUILD_TESTS:                ${XEUS_BUILD_TESTS}")  
+message(STATUS "XEUS_USE_DYNAMIC_UUID:           ${XEUS_USE_DYNAMIC_UUID}") 
 
 # Dependencies
 # ============
@@ -172,12 +174,14 @@ macro(xeus_create_target target_name linkage output_name)
         if (APPLE)
             target_link_libraries(${target_name} PUBLIC "-framework CoreFoundation")
         else ()
-            if (XEUS_STATIC_DEPENDENCIES)
+            if ((NOT XEUS_USE_DYNAMIC_UUID) AND (XEUS_STATIC_DEPENDENCIES))
+                message(STATUS "static libuuid linking")
                 find_path(LIBUUID_INCLUDE_DIR uuid/uuid.h)
                 find_library(LIBUUID_LIBRARY libuuid.a)
                 target_include_directories(${target_name} PRIVATE ${LIBUUID_INCLUDE_DIR})
                 target_link_libraries(${target_name} PRIVATE ${LIBUUID_LIBRARY}) 
             else ()
+                message(STATUS "dynamic libuuid linking")
                 find_package(LibUUID REQUIRED)
                 target_link_libraries(${target_name} PRIVATE LibUUID::LibUUID)
             endif ()
-- 
2.43.0

